src/AggregateRoots/AggregateRoot.php

```diff
@@ -129,6 +157,42 @@ public function persist(): static
         return $this;
     }
 
+    public function persistConcurrently(int $tries = -1): static
```

Comment:
Would it be possible to use `persist()` as a shorthand if you don't want to specify tries? That way there'll be less mental overhead for the common use case.

Reply:
This is what I was building initially. However, when the aggregate root is re-retrieved after a concurrency exception, it has a new reference because it's a new object. Unfortunately there's no way to keep the original AR object without heavy use of reflection and black PHP magic.

That would be a breaking change for the `persists` method. This could cause issues:

```php
$aggregateRoot = AggregateRoot::retrieve(…);
// …

$aggregateRoot->persist();
$aggregateRoot->doSomethingElse();
```

To fix:

```php
$aggregateRoot = AggregateRoot::retrieve(…);
// …

$aggregateRoot = $aggregateRoot->persist();
$aggregateRoot->doSomethingElse();
```

I think this is too risky of a change to introduce in a minor version. In the next major version, I'd like to address this and provide a single API for concurrently recording events (both for `record`/`recordConcurrently` and `persist`/`persistConcurrently`

Reply:
Gotcha, it makes sense then to keep them separate :)